salt/_modules/metalk8s_network.py

```diff
@@ -167,3 +167,58 @@ def get_listening_processes():
         )
 
     return all_listen_connections
+
+
+def routes():
+    """
+    Return currently configured IPv4 routes from routing table
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt '*' metalk8s_network.routes
+    """
+    ret = []
+    cmd = "ip -4 route show table main"
+    out = __salt__["cmd.run"](cmd)
+    for line in out.splitlines():
+        comps = line.split()
+
+        # ignore blackhole routes.
+        if comps[0] in ("unreachable", "blackhole"):
+            continue
+
+        if comps[0] == "default":
+            ip_interface = ""
+            if comps[3] == "dev":
+                ip_interface = comps[4]
+
+            ret.append(
+                {
+                    "addr_family": "inet",
+                    "destination": "0.0.0.0",
+                    "gateway": comps[2],
+                    "netmask": "0.0.0.0",
+                    "flags": "UG",
+                    "interface": ip_interface,
+                }
+            )
+        else:
+            address_mask = __salt__["network.convert_cidr"](comps[0])
```

Comment:
So, if for some reason we miss another special case (I don't know if we can be 100% certain that the first part of a line is always either `unreachable`, `blackhole`, `default`, or a valid CIDR), this will fail just as before with `blackhole`.

Wouldn't it make sense to catch errors here, and handle it with an error log instead of crashing? I mean, all we care about is whether a valid route is available, not checking the sanity (if there is such a thing) of the overall routing table.